### PR TITLE
Add OllamaClient implementation of AiClient

### PR DESCRIPTION
**Pull request by lemon0029**
Hi, I just implemented the Ollama LLM client. Can someone review it?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,6 +22,7 @@
 		<module>vector-stores/spring-ai-pgvector-store</module>
 		<module>vector-stores/spring-ai-milvus-store</module>
 		<module>vector-stores/spring-ai-neo4j-store</module>
+		<module>spring-ai-ollama</module>
```
Comment (eddumelendez) [inline pom.xml:25]: starter for ollama should be provided too

File: spring-ai-ollama/src/main/java/org/springframework/ai/ollama/client/OllamaClient.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    private String getGenerateRequestBody(String prompt) {
+        var data = Map.of(
+                "model", model,
+                "prompt", prompt
+        );
+
+        try {
+            return jsonMapper.writeValueAsString(data);
+        } catch (JsonProcessingException ignored) {
+        }
```
Comment (markpollack) [inline spring-ai-ollama/src/main/java/org/springframework/ai/ollama/client/OllamaClient.java:105]: need to rethrow runtimeexception

File: spring-ai-ollama/src/main/java/org/springframework/ai/ollama/client/OllamaClient.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+    private static final HttpClient httpClient = HttpClient.newBuilder().build();
+
+    private final String baseUrl;
+
+    private final String model;
+
+    private Consumer<OllamaGenerateResult> simpleCallback;
+
+    public OllamaClient(String baseUrl, String model) {
```
Comment (markpollack) [inline spring-ai-ollama/src/main/java/org/springframework/ai/ollama/client/OllamaClient.java:45]: add javadocs to ctor, methods, etc.

**Review by markpollack — COMMENTED** [review]
will make changes on merge.  Thanks for the PR.  If you can, creating the starter would be much appreaciated.

**Comment by markpollack** [comment]
will make changes on merge. Thanks for the PR. If you can, creating the starter would be much appreaciated.

**Comment by markpollack** [comment]
Merged as e68bdeb9a0d74badacaccc1677521fd2612f5b71

**Pull request closed without merging**